src/status_im2/contexts/chat/messages/content/reactions/view.cljs

```diff
@@ -5,31 +5,79 @@
             [utils.re-frame :as rf]
             [status-im2.contexts.chat.messages.drawers.view :as drawers]))
 
+(defn- on-press
+  [own message-id emoji-id emoji-reaction-id]
+  (if own
+    (rf/dispatch [:models.reactions/send-emoji-reaction-retraction
+                  {:message-id        message-id
+                   :emoji-id          emoji-id
+                   :emoji-reaction-id emoji-reaction-id}])
+    (rf/dispatch [:models.reactions/send-emoji-reaction
+                  {:message-id message-id
+                   :emoji-id   emoji-id}])))
+
+(defn- on-long-press
+  [chat-id message-id emoji-id show-reaction-authors-sheet?]
+  (reset! show-reaction-authors-sheet? true)
+  (rf/dispatch [:chat.ui/emoji-reactions-by-message-id
+                {:message-id         message-id
+                 :chat-id            chat-id
+                 :long-pressed-emoji emoji-id}]))
+
+(defn message-reactions-row-comp
+  [{:keys [message-id chat-id]} user-message-content show-reaction-authors-sheet?]
+  (let [reactions                                 (rf/sub [:chats/message-reactions message-id chat-id])
+        [reaction-authors-list
+         reaction-authors-list-selected-reaction] (rf/sub [:message/reaction-authors-list chat-id])
+        _show-sheet                               (when (and reaction-authors-list
+                                                             @show-reaction-authors-sheet?)
+                                                    (rf/dispatch [:dismiss-keyboard])
+                                                    (rf/dispatch
+                                                     [:show-bottom-sheet
+                                                      {:on-close
+                                                       (fn []
+                                                         (reset! show-reaction-authors-sheet? false)
+                                                         (rf/dispatch
+                                                          [:chat/clear-emoji-reaction-author-details]))
+                                                       :content
+                                                       (fn [] [drawers/reaction-authors
+                                                               reaction-authors-list
+                                                               reaction-authors-list-selected-reaction])
+                                                       :selected-item (fn []
+                                                                        user-message-content)
+                                                       :padding-bottom-override 0}]))]
+    [:<>
+     (when (seq reactions)
+       [rn/view
+        {:style {:margin-left    52
+                 :margin-bottom  12
+                 :flex-direction :row}}
+        (for [{:keys [own emoji-id quantity emoji-reaction-id]
+               :as   emoji-reaction} reactions]
+          ^{:key (str emoji-reaction)}
```

Comment:
Not something from your PR, but it's worth sharing:

If the `emoji-id` is unique, shouldn't we be using it instead of stringifying the whole `emoji-reaction`? In general, I think we should just avoid using `str` on data structures as a means of creating unique keys, it's inefficient in many cases.

Reply:
By inefficient do you mean it's unnecessary?

Reply:
Sorry for not being more specific. Inefficient because building the string representation of a data structure requires significant computation compared to simply using a good/known key, like `emoji-id`. Also the key will be larger in memory and put more pressure on the GC. We should just avoid this, especially for larger data structures. Maybe for a reaction is fine, but why go into the trouble you know what I mean? It's better to just avoid using `str` on data structures.

Reply:
FYI @flexsurfer and I were talking about starting a documentation on good performance practices for `status-mobile`. Among so many more important performance considerations, the point I shared above should be there imo, even though it's of less significance/impact.